src/consensus/pbft/libbyz/test/test_ledger_replay.cpp

```diff
@@ -145,6 +162,21 @@ TEST_CASE("Test Ledger Replay")
       size_t num_requests = 1;
       auto pp = std::make_unique<Pre_prepare>(1, i, rqueue, num_requests);
 
+      ccf::Store::Tx tx;
+      auto req_view = tx.get_view(write_pbft_requests);
+      req_view->put(
+        0,
+        {0,
+         0,
+         {},
+         {(const uint8_t*)request->contents(),
+          (const uint8_t*)request->contents() + request->size()}});
+
+      auto der_view = tx.get_view(write_derived_map);
+      der_view->put("key1", "value1");
+
+      REQUIRE(tx.commit() == kv::CommitSuccess::OK);
+
       // imitate exec command
```

Comment:
How difficult would it be to actually have this work done in `ExecutionMock`'s `exec_command()`? IMO, it would give a much more accurate test (i.e. actually relying on the calls to `ledger_writer->write_pre_prepare()` in `Replica.cpp`), instead of having to play with the KV's interface.

Reply:
Not sure this has to do with work done in `exec_command`. to have `Replica` call `ledger_writer->write_pre_prepare()` we need a full run of the protocol where requests are sent and are processed via pre-prepare, prepares, etc (like the `test_UDP` end to end test for example). Here we just want to populate the ledger with requests and pre-prepares so that we unit test playing back the ledger further down.

Reply:
Changing to trigger `ledger_writer->write_pre_prepare()` via `Replica::handle(Request)` as discussed, instead of manually creating ad writing the pre-prepares.